bundler/lib/bundler/fetcher/gem_remote_fetcher.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require "rubygems/remote_fetcher"
+
+module Bundler
+  class Fetcher
+    class GemRemoteFetcher < Gem::RemoteFetcher
+      def request(*args)
+        super do |req|
+          req.delete("User-Agent") if headers["User-Agent"]
+          yield req if block_given?
+        end
+      end
+    end
+  end
+end
```

Comment:
Why are we adding this class? If we use a dedicated fetcher instance per remote and no longer use `Gem::RemoteFetcher.fetcher`, do we still need the cleanup?

Reply:
needed because `Gem::RemoteFetcher` only `add`s the supplied headers, and we need to be able to completely _set_ the UA header, otherwise all the parsing we do of bundler UAs won't work

Reply:
I don't fully understand. Previously we were using a single instance and were using the `headers` setter to set headers, right? Was that incorrect already? Is there any test that fails if we use don't add this class, in case that illustrates its need?

Reply:
Yes, the artifice test. The header needs to be cleared so the custom UA is set rather than appended to the rubygems UA

Reply:
I don't know for sure that this class is the best approach, but I feel confident that it won't be difficult to refactor later, if needed, since it adds no new interface. We can revisit, but I think this fix should go in so the parallel fetching code is not broken on master.

Reply:
Cool, cool, I'm just curious but did not mean to block this fix.